File: 19. Remove Nth Node From End of List/19.solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
  public ListNode removeNthFromEnd(ListNode head, int n) {
    ListNode dummyListNode = new ListNode(0);
    dummyListNode.next = head;
    int length = 0;
    ListNode first = head;
    
    while (first != null) {
      length++;
      first = first.next;
    }
    
    length -= n;
    first = dummyListNode;
    
    while (length > 0) {
      length--;
      first = first.next;
    }
    
    first.next = first.next.next;
    
    return dummyListNode.next;
  }
}

File: 19. Remove Nth Node From End of List/19.solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function (head, n) {
  const dummyListNode = new ListNode(0);
  dummyListNode.next = head;
  let length = 0;
  let first = head;

  while (first) {
    length++;
    first = first.next;
  }

  length -= n;
  first = dummyListNode;

  while (length > 0) {
    length--;
    first = first.next;
  }

  first.next = first.next.next;

  return dummyListNode.next;
};
